tests/build_/test_build_request.py

```diff
@@ -1525,6 +1523,9 @@ def test_render_orchestrate_build(self, tmpdir, platforms, secret, disabled,
             kwargs['koji_parent_build'] = koji_parent_build
         if prefer_schema1_digest is not None:
             kwargs['prefer_schema1_digest'] = prefer_schema1_digest
+        # required for pulp_ plugins in the orchestrator build
+        if arrangement_version > 3:
+            kwargs['pulp_secret'] = 'pulp_secret'
```

Comment:
I'm assuming this is needed because orchestrator build uses pulp registry starting on arrangement version 4, and whenever a `pulp_registry_name` is provided, `pulp_secret` is required.

What's puzzling me is that I don't see how `pulp_registry_name` is being set for this test run which make me wonder if we have an incorrect check somewhere.

Reply:
Aha! It does get set by `additional_kwargs` param.
It didn't complain before because pulp registry was actually never used by orchestrator build.
I think `pulp_secret` should be set alongside it, around line 1469.